esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp

```diff
@@ -65,6 +65,9 @@ void ESP32BLETracker::setup() {
       [this](ota::OTAState state, float progress, uint8_t error, ota::OTAComponent *comp) {
         if (state == ota::OTA_STARTED) {
           this->stop_scan();
+          for (auto *client : this->clients_) {
+            client->set_state(ClientState::DISCONNECTING);
+          }
```

Comment:
This is probably all that is needed for this entire PR tbh. All of the bluetooth proxy clients are in the `clients_` list.
```suggestion
          for (auto *client : this->clients_) {
            client->disconnect();
          }
```

Reply:
So that was my original plan, but there is no public `disconnect` method it seems, which explains why I had trouble finding it ...

From the CI tests:

```
### File esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp

Error: /home/runner/work/esphome/esphome/esphome/components/esp32_ble_tracker/esp32_ble_tracker.cpp:69:21: error: no member named 'disconnect' in 'esphome::esp32_ble_tracker::ESPBTClient'; did you mean 'connect'? [clang-diagnostic-error]
            client->disconnect();
                    ^~~~~~~~~~
                    connect
/home/runner/work/esphome/esphome/esphome/components/esp32_ble_tracker/esp32_ble_tracker.h:[17](https://github.com/esphome/esphome/actions/runs/11603000836/job/32309288708?pr=7695#step:9:18)4:16: note: 'connect' declared here
  virtual void connect() = 0;
```

Reply:
OH, its on an intermediate class.

Maybe we can fix that with #7705. Try merging that branch into this one

Reply:
Compiles! I'm hoping to test it tomorrow

Reply:
How did this work out for you? Does it help / solve the problem?

Reply:
Sorry for the delay! I've been using this at home for a bit now with great success. I combined it with my attempts at adding bluetooth proxy sensors and I have pretty good stability with my bluetooth proxies during OTA, with insight